### DISCUSSION TITLE
KRP928B2S connection

### DISCUSSION THREAD

**Discussion by jchenny7** [revk/xESP32-Faikout#247] [General]
Is it possible to connect Faikin to KRP928B2S?
S21 is already occupied by KRP928B2S to connect to a central control panel, is there any port on KRP928B2S I can connect Faikin to, or do I have to disconnect KRP928B2S and sacrifice the central control panel?

Thank you.
![1708772741638](https://github.com/revk/ESP32-Faikin/assets/114911805/09f57a16-8c72-463a-80c9-d3ce6fd09704)
![22143699738318_456](https://github.com/revk/ESP32-Faikin/assets/114911805/7a6e2e86-e578-4f34-971b-5e0b73f402c6)

**Comment by MUN0X** [comment]
Cant you connect faikin to s19. Not sure but I think that BRC944, that is connected to it in the schematic, can be connected to s21 directly. So s19 might be in fact a s21 port.
Otherwise, I believe I came across  s21 splitters in my search for alternatives for the genuine daikin modules.

Edit: [https://shop.prestige-distribution.fr/en_GB/shop/cau-s21-splitter-for-dk-s21-splitter-for-dk-1089](url)

**Reply by njh** [reply]
I wonder how a S21 splitter would work - it doesn't look like there is much electronics there. As it is just a basic serial protocol, I would expect them just to talk over each other. Unless there is some kind of listen/back-off implemented?

**Reply by jchenny7** [reply]
S19 connects to my all control panel, without it my all contorl panel would not be funactionable. 
I thought about making direct connection to S19, but when I read through the installation manual of BRC944, it shows the connection from BRC944 needs to goto a remote control adpater before it can connect to S21 on the machine, or do you think S19 should be workable? the Pin/connector for S19 seems to be slightly different to S21, some modification might be needed on the connector. 
Appreciate for your help. 

<img width="492" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/114911805/5d534898-fce7-4486-825a-67012cce4132">

**Reply by jchenny7** [reply]
I do not have great electrician knowledge, how difficult is it to make a splitter as I would assume there is nothing readily avaialble in the market, could you kindly lead me through on how and the safest way to do it? would there be any potential risk on doing it?
Thank you.

**Reply by revk** [reply]
I am not sure how a "splitter" would work for the serial data used on a normal S21. I mean it is conceivable that a remote might work in-line in a loop some how passing on messages from the other remote or something, but that would be a very different way of working. I guess that could allow a simple electrical splitter though. No clue.

As do S19, I don't know, but there was mention of a port that needs an adapter to make it S21, and I think the reason was that the port was not electrically isolated. Ie it worked at mains voltages. I know someone managed to make the port work with a Faikin, but meaning the Faikin was live, so carefully fitted inside the housing as a result. I would not recommend playing with mains unless you really know what you are doing! So very much at your own risk.

**Comment by genehilton** [comment]
The splitter sounds like more of a "breakout" cable.  It might be very useful for snooping/debugging an S21 connection, but S21 doesn't seem anything like RS485 where you can hang multiple receivers/transmitters on the same wire pair.

I'm lurking here so far as the Faikin seems fantastic, but the condo I just moved into has 3x BRC944B2 wired controllers for the 3 FTXR12TVJUW indoor units.  I haven't opened up the indoor units to look at the PCBs yet, but my manual reading makes me think in my case it's either the BRCs or Faikin (only one S21 connector).  The Faikin is so much better, but then I would have a powered off BRC hanging on my wall.

**Reply by jchenny7** [reply]
That is exacting the same problem I am having... I am seeking an option so that I can still make use of the BRC944B2 while also having Faikin installed....

**Reply by revk** [reply]
The S21 seems to have open drain drive and pullups, so in theory it could multi-drop, but is that really what they do? Not out of the question I guess!

**Reply by jchenny7** [reply]
Coz I have been spoken to a few Daikin engineers, and none of them thinks the S21 splitter would work... I want to give it a go, but unfortunately there is only one supplier and it's in EU, and not interested to ship to Asia ..

**Reply by genehilton** [reply]
[This page](https://support.coolautomation.com/hc/en-us/article_attachments/16843754323485) seems to imply  the splitter might work, although in their newer models they moved the splitting functionality to their "Cool Plug".  There is also a [place](https://blvs.com/customer-portal/products/home-automation/home-automation-devices/s21-splitter-for-dk/) in the US that might sell it too.  Could you use a [forward2me](https://forward2me.com/) or some other forwarding service?

**Reply by jchenny7** [reply]
Thanks for sharing the sites, but the site seems to do wholesales only, don't think they are interested to sell 3 sets only... I do have some courier service able to forward the parcel from US to Asia, but needs to find a place to buy.
